stan/math/opencl/triangular.hpp

```diff
@@ -0,0 +1,58 @@
+#ifndef STAN_MATH_OPENCL_TRIANGULAR_HPP
+#define STAN_MATH_OPENCL_TRIANGULAR_HPP
+#ifdef STAN_OPENCL
+namespace stan {
+namespace math {
+enum class TriangularViewCL { Diagonal = 0, Lower = 1, Upper = 2, Entire = 3 };
```

Comment:
Should we change the name of this to `TriangularTypeCL` or something like that? We could probably remove the CL part

Reply:
We still store whole matrix and this attribute only determines which part most functions view. So I prefere view to type. I also like that it has something OpenCL-related in its name. If we move it to some cl-related namespace than im fine with dropping CL from the name.

Reply:
Still gotta do this, Maybe just `SpecialViewsCL` or does Eigen have these sorted out already and we can just steal those?